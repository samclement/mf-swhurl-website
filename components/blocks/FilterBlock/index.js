import React from 'react'
import Link from 'next/link'
import styled from 'styled-components'

const StyledBlock = styled.div`
  width: 170px;
  padding: 10px;
`

const StyledLink = styled.a`
  color: black;
  font-size: 12px;
  text-decoration: none;
  display: block;
  padding-left: 10px;
  text-indent: -15px;
  margin-bottom: 6px;
  ::before {
    content: ${props =>
      props.selected
        ? 'url("/static/checkbox_checked.svg")'
        : 'url("/static/checkbox_unchecked.svg")'};
  }
`

export default ({ title, filters }) => {
  return (
    <StyledBlock>
      <h2>{title}</h2>
      {filters.map((f, i) => {
        const url = `/plp?url=${f.url}`
        return (
          <Link prefetch key={i} href={url} as={f.url}>
            <StyledLink href="#" selected={f.selected}>
              {f.name}
            </StyledLink>
          </Link>
        )
      })}
    </StyledBlock>
  )
}
