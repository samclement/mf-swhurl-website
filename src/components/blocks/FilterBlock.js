import React from 'react'
import Link from 'next/link'
import styled from 'styled-components'

const StyledBlock = styled.div`
  width: 210px;
  padding: 10px;
`

const StyledFilters = styled.div`
  max-height: ${props => (props.scrollable ? '420px' : 'initial')};
  overflow-y: ${props => (props.scrollable ? 'auto' : 'initial')};
  border: ${props => (props.scrollable ? '1px solid #ccc' : 'initial')};
  padding: ${props => (props.scrollable ? '5px 9px' : 'initial')};
`

const StyledLink = styled.a`
  color: black;
  font-size: 13px;
  text-decoration: none;
  display: block;
  padding-left: 10px;
  text-indent: -15px;
  margin-bottom: 6px;
  ::before {
    content: ${props =>
      props.selected
        ? 'url("/checkbox_checked.svg")'
        : 'url("/checkbox_unchecked.svg")'};
    position: relative;
    bottom: -2px;
  }
`

export default ({ title, filters, scrollable }) => {
  return (
    <StyledBlock>
      <h2>{title}</h2>
      <StyledFilters scrollable={scrollable}>
        {filters.map((f, i) => {
          const url = `/plp?url=${f.url}`
          return (
            <Link key={i} href={url} as={f.url}>
              <StyledLink href={f.url} selected={f.selected}>
                {f.name}
              </StyledLink>
            </Link>
          )
        })}
      </StyledFilters>
    </StyledBlock>
  )
}
