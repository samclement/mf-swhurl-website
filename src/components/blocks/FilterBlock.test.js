/* eslint-env jest */

import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import FilterBlock from './FilterBlock.js'
// Data
import data from '../../../test-data/activewear.json'
import { parseSearchResults } from '../../lib/parser.js'
// Nextjs routing support
import Router from 'next/router'
const mockedRouter = { push: () => {} }
Router.router = mockedRouter

const facets = parseSearchResults(data).facets
const facet = facets[0]

describe('With Snapshot Testing', () => {
  it('FilterBlock displays items', () => {
    const { asFragment } = render(
      <FilterBlock title={facet.name} filters={facet.values} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
