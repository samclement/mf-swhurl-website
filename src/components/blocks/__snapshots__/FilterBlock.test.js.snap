// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing FilterBlock displays items 1`] = `
<DocumentFragment>
  .c0 {
  width: 210px;
  padding: 10px;
}

.c1 {
  max-height: initial;
  overflow-y: initial;
  border: initial;
  padding: initial;
}

.c2 {
  color: black;
  font-size: 13px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  padding-left: 10px;
  text-indent: -15px;
  margin-bottom: 6px;
}

.c2::before {
  content: url("/checkbox_unchecked.svg");
  position: relative;
  bottom: -2px;
}

<div
    class="c0"
  >
    <h2>
      Designer
    </h2>
    <div
      class="c1"
    >
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensalyx"
      >
        1017 ALYX 9SM
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amens-2xu"
      >
        2XU
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensandwander"
      >
        And Wander
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensaztech-mountain"
      >
        Aztech Mountain
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensblackbarrett-by"
      >
        Blackbarrett by Neil Barrett
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenscafe-du"
      >
        Caf√© Du Cycliste
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenscastore"
      >
        Castore
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensfalke"
      >
        Falke
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensfendi"
      >
        Fendi
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenshelly-hansen"
      >
        Helly Hansen
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensiffley-road"
      >
        Iffley Road
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenslndr"
      >
        LNDR
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensmammut"
      >
        Mammut Delta X
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensmoncler"
      >
        Moncler
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensmoncler-genius"
      >
        Moncler Genius
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensmoncler-grenoble"
      >
        Moncler Grenoble
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensnemen"
      >
        Nemen
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenspeak-performance"
      >
        Peak Performance
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensperfect-moment"
      >
        Perfect Moment
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenssalomon"
      >
        Salomon
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenssatisfy"
      >
        Satisfy
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amensthe-upside"
      >
        The Upside
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing/activewear?q=%3A%3AallDesigners%3Amenstilak"
      >
        Tilak
      </a>
    </div>
  </div>
</DocumentFragment>
`;
